core/src/main/java/be/isach/ultracosmetics/cosmetics/gadgets/GadgetBlackHole.java

```diff
@@ -41,7 +40,7 @@ protected void onRightClick() {
 
         item = ItemFactory.spawnUnpickableItem(XMaterial.BLACK_TERRACOTTA.parseItem(), getPlayer().getEyeLocation(), getPlayer().getEyeLocation().getDirection().multiply(1.3d));
 
-        Bukkit.getScheduler().runTaskLater(getUltraCosmetics(), () -> {
+        getUltraCosmetics().getScheduler().runAtEntityLater(getPlayer(), () -> {
```

Comment:
As with firework, `runAtEntityLater(item, ...`?

Reply:
Need to be careful with scheduling around entities that can despawn, location is more appropriate here on that basis. Refactored!